feat(app): add e-stop modals for odd

koji opened the pull request:
<!--
Thanks for taking the time to open a pull request! Please make sure you've read the "Opening Pull Requests" section of our Contributing Guide:

https://github.com/Opentrons/opentrons/blob/edge/CONTRIBUTING.md#opening-pull-requests

To ensure your code is reviewed quickly and thoroughly, please fill out the sections below to the best of your ability!
-->

# Overview
Add two modals for e-stop. One is EstopPressedModal and the other is EstopMissingModal.
Combine DesktopEstopPressedModal and EstopPressedModal into one component as well as DesktopMissingModal.
Also add `error` type to Chip component.
The function will be added by following PRs.

design
https://www.figma.com/file/t4GKKfUA3RRfgHFxPbVVjV/Release%3A-Touchscreen-August-update?type=design&node-id=629-52540&mode=design&t=ucnIizoesp9Fg9hI-0

<!--
Use this section to describe your pull-request at a high level. If the PR addresses any open issues, please tag the issues here.
-->

# Test Plan
1. run storybook
check EstopPressedModal and EstopMissingModal under `Desktop` and `ODD`
<!--
Use this section to describe the steps that you took to test your Pull Request.
If you did not perform any testing provide justification why.

OT-3 Developers: You should default to testing on actual physical hardware.
Once again, if you did not perform testing against hardware, justify why.

Note: It can be helpful to write a test plan before doing development

Example Test Plan (HTTP API Change)

- Verified that new optional argument `dance-party` causes the robot to flash its lights, move the pipettes,
then home.
- Verified that when you omit the `dance-party` option the robot homes normally
- Added protocol that uses `dance-party` argument to G-Code Testing Suite
- Ran protocol that did not use `dance-party` argument and everything was successful
- Added unit tests to validate that changes to pydantic model are correct

-->

# Changelog
- add EstopPressedModal and tests
- add EstopMissingModal and tests
- add `error` type to Chip component and test
<!--
List out the changes to the code in this PR. Please try your best to categorize your changes and describe what has changed and why.

Example changelog:
- Fixed app crash when trying to calibrate an illegal pipette
- Added state to API to track pipette usage
- Updated API docs to mention only two pipettes are supported

IMPORTANT: MAKE SURE ANY BREAKING CHANGES ARE PROPERLY COMMUNICATED
-->

# Review requests

<!--
Describe any requests for your reviewers here.
-->

# Risk assessment
low
<!--
Carefully go over your pull request and look at the other parts of the codebase it may affect. Look for the possibility, even if you think it's small, that your change may affect some other part of the system - for instance, changing return tip behavior in protocol may also change the behavior of labware calibration.

Identify the other parts of the system your codebase may affect, so that in addition to your own review and testing, other people who may not have the system internalized as much as you can focus their attention and testing there.
-->

koji committed:
feat(app): add e-stop modals for odd

add two modals for e-stop. One is EstoPressedModal and the other is EstopMissingModal.

koji committed:
add tests

koji committed:
add error type tests to Chip component

koji committed:
Merge branch 'edge' into feat_add-estop-modals-for-touchscreen

koji committed:
update json file

koji commented:
- [ ] combine DesktopEstopPressedModal + EstopPressedModal (ODD) ->  EstopPressedModal
- [ ] combine DesktopEstopMissingModal + EstopMissingModal (ODD) ->  EstopMissingModal

koji committed:
combine estop modals for desktop and touchscreen

koji committed:
update tests for estop modals

jerader commented on app/src/atoms/Chip/__tests__/Chip.test.tsx:
```diff
@@ -110,4 +110,33 @@ describe('Chip', () => {
+    expect(chip).toHaveStyle(`background-color: ${String(COLORS.red3)}`)
+    expect(chipText).toHaveStyle(`color: ${String(COLORS.red1)}`)
+    const icon = getByLabelText('icon_mockError')
+    expect(icon).toHaveStyle(`color: ${String(COLORS.red1)}`)
```
Is adding the String here necessary?

```suggestion
    expect(icon).toHaveStyle(`color: ${COLORS.red1}`)
```

jerader commented on app/src/atoms/Chip/__tests__/Chip.test.tsx:
```diff
@@ -110,4 +110,33 @@ describe('Chip', () => {
+    expect(chip).toHaveStyle(`background-color: ${String(COLORS.transparent)}`)
+    expect(chipText).toHaveStyle(`color: ${String(COLORS.red1)}`)
+    const icon = getByLabelText('icon_mockError')
+    expect(icon).toHaveStyle(`color: ${String(COLORS.red1)}`)
```
same here

jerader commented on app/src/organisms/EstopModal/EstopPressedModal.tsx:
```diff
@@ -0,0 +1,161 @@
+          width="100%"
+          buttonText={t('resume_robot_operations')}
+          disabled={isEngaged}
+          onClick={() => console.log('pressed')}
```
you can leave a todo comment here!

jerader commented on app/src/organisms/EstopModal/__tests__/EsoptPressedModal.test.tsx:
```diff
@@ -0,0 +1,123 @@
+  })
+
+  it('should render text and button - inactive run', () => {
+    props.isActiveRun = false
```
wow, i didn't realize you can reinitialize props this way, that's cool!

jerader reviewed (approved):
just left some tiny cleanup comments but this lgtm!

koji committed:
address comments

shlokamin commented on app/src/atoms/Chip/index.tsx:
```diff
@@ -43,6 +43,12 @@ const CHIP_PROPS_BY_TYPE: Record<
     borderRadius: BORDERS.borderRadiusSize1,
     textColor: COLORS.darkBlack90,
   },
+  error: {
+    backgroundColor: COLORS.red3,
+    borderRadius: BORDERS.borderRadiusSize6,
+    iconColor: COLORS.red1,
+    textColor: COLORS.red1,
+  },
```
i think this makes sense as a chip (rather than a `ListItem` like Figma says), but the chip's border radius is 24px i think: https://www.figma.com/file/OIdG64Q5cgvEw82ish5Eee/Design-System-(Flex)?type=design&node-id=1121%3A61901&mode=design&t=Wb9HwO708RGc6znQ-1

shlokamin replied:
should we update the designs to have the e stop modal use a chip rather than list item? cc @mmencarelli does Rob have a github? i can't seem to find it

shlokamin replied:
i think its fine to implement as a chip, but let's have the chip's styles match what's in Figma

koji replied:
as we talked, we use BORDERS.borderRadiusSize5: 40px for now.

koji committed:
update the modal config to align with the latest design

no stroke for e-stop modals
for e-stop missing modal of the touchscreen app, the size was changed into medium from large

koji committed:
update test cases for the design update

koji committed:
add border-radius check to Chip tests

koji committed:
Merge branch 'edge' into feat_add-estop-modals-for-touchscreen

koji committed:
fix linting error
